### PR TITLE
[NEP-779] Add VC_Saved event

### PR DESCRIPTION
**Pull request by everdark**
https://replika.atlassian.net/browse/NEP-779

Related to:

+ https://github.com/affinityproject/affinity-metrics-lib/pull/11
+ https://github.com/affinityproject/affinity-metrics/pull/77

### CODE REVIEW

File: sdk/browser/src/AffinityWallet.ts [hunk 1]
```diff
@@ -157,6 +157,9 @@ export class AffinityWallet extends CoreNetwork {
   async saveCredentials(data: any): Promise<any> {
     const encryptedCredentials = await this.walletStorageService.encryptCredentials(data)
 
+    this._sendVCSavedMetrics(data) // encryption and saving are both async, saveCredentials returns a promise.
+                                   // we should only send metrics once it is actually saved, but how?
+
     return this.saveEncryptedCredentials(encryptedCredentials)
```
Comment (rbrazhnyk) [inline sdk/browser/src/AffinityWallet.ts:163]: we can have something like:

```ts
  const result = await this.saveEncryptedCredentials(encryptedCredentials)

  this._sendVCSavedMetrics(data)
  // NOTE: what if creds actually were not saved in the vault? Follow up with Isaak/Dustin on this - should we parse the response to define if we need to send the metrics

  return result
```

File: sdk/core/src/CommonNetworkMember.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { buildVCV1Unsigned, buildVCV1Skeleton, buildVPV1Unsigned } from '@affini
 import { VCV1, VCV1SubjectBaseMA, VPV1, VCV1Unsigned } from '@affinidi/vc-common'
 import { parse } from 'did-resolver'
 
-import { EventCategory, EventName } from '@affinityproject/affinity-metrics-lib'
+import { EventCategory, EventName, VcMetadata } from '@affinityproject/affinity-metrics-lib'
```
Comment (rbrazhnyk) [inline sdk/core/src/CommonNetworkMember.ts:7]: as agreed, let's have `Metadata` in metrics-lib

File: sdk/core/src/CommonNetworkMember.ts [hunk 3]
```diff
@@ -1834,6 +1834,36 @@ export class CommonNetworkMember {
...
+
+  protected _sendVCSavedMetrics(credentials: any) {
+    for (const credential of credentials) {
+      const vcMetadata = {vcType: credential.type}
+      const vcId = credential.id
+      // the issuer property could be either an URI string or an object with id propoerty
+      // https://www.w3.org/TR/vc-data-model/#issuer
+      let issuerId: string
+      const issuer = credential.issuer
+      if (typeof issuer === 'string') {
```
Comment (rbrazhnyk) [inline sdk/core/src/CommonNetworkMember.ts:1858]: please check if there would not be any linter issues. `npm run lint`

**Review by rbrazhnyk — CHANGES_REQUESTED** [review]
Please see minor comments

**Review by rbrazhnyk — APPROVED** [review]
@everdark, adding approval. Please make sure event is saved before merging. Ping me if you need any help here.

**Review by rbrazhnyk — DISMISSED** [review]
Please merge metrics-lib, update metrics backend, and after all checks in the pipeline passed there feel free to merge.

**Review by ipatka — APPROVED** [review]

**Pull request merged**